components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu
+
+import android.content.Context
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+
+/**
+ * Browser menu builder with web extension support.
+ *
+ * @param store [BrowserStore] required to render web extension browser actions
+ * @param appendExtensionAtStart true if web extensions appear at the top (start) of the menu,
+ * false if web extensions appear at the bottom of the menu. Default to false (bottom).
+ */
+class WebExtensionBrowserMenuBuilder(
+    items: List<BrowserMenuItem>,
+    extras: Map<String, Any> = emptyMap(),
+    endOfMenuAlwaysVisible: Boolean = false,
+    private val store: BrowserStore,
+    private val appendExtensionAtStart: Boolean = false
```

Comment:
`appendExtensionActionsAtStart` or similar to indicate what we're appending. It's not the extensions...